packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -293,8 +336,28 @@ export class BucketDeployment extends CoreConstruct {
       uuid += `-${memoryLimit.toString()}MiB`;
     }
 
+    // if user specify to use VPC, define another singleton handler
+    // with this configuration. otherwise, it won't be possible to use multiple
+    // configurations since we have a singleton.
+    // A VPC is a must if EFS storage is used and that's why we are only using VPC in uuid.
+    if (vpc) {
+      uuid += `-${vpc.node.addr.toUpperCase()}`;
```

Comment:
```suggestion
      uuid += `-${vpc.node.addr}`;
```

Altering the case of the `vpc.node.addr` may reduce the entropy of if, and increase the risks of collision.

Reply:
Updated as suggested.